src/ffi.rs

```diff
@@ -38,6 +38,9 @@ use libc::c_void;
 use libc::size_t;
 use libc::ssize_t;
 
+#[cfg(unix)]
```

Comment:
I think this is ok to be specific to linux? (I don't know other unix-like OS has similar support of SO_TXTIME)

Reply:
Yes, I think it should be good for now.

Reply:
This will break on macOS, as `std::time::Instant` doesn't use `struct timespec`, so it needs to be excluded I think, see https://github.com/rust-lang/rust/blob/master/library/std/src/sys/unix/time.rs#L274

Reply:
this is to include libc::timespec struct. I added unix check to differentiate between unix family and windows family. Windows does not have timespec
https://doc.rust-lang.org/reference/conditional-compilation.html#target_family

Reply:
I think we still need to handle MacOS/iOS case as @ghedo mentioned above.

Reply:
How about adding 'linux only' check here

Reply:
sounds good to me.